README.md

```diff
@@ -1,87 +1,135 @@
+
+    ```bash
+    tune_rst_parser rst_discourse_tb_edus_TRAINING_TRAIN.json rst_discourse_tb_edus_TRAINING_DEV.json rst_parsing_model
+    ```
+
+    This command iterates over a pre-defined list of values for the `C` regularization parameter for logistic regression, trains a model using the features extracted from the training set, and then evaluates that model on the development set. Its final output is the `C` value that yields the highest performance F1 score on the development set. After this command, you will have a number of directories with the prefix `rst_parsing_model` in the current directory, e.g., `rst_parsing_model.C0.25`, `segmentation_model.C1.0` et cetera. Each of these directories contains the logistic regression model files (named `rst_parsing_all_feats_LogisticRegression.model`) trained with those specific values of the `C` regularization parameter.  Underlyingly, this command uses the [SKLL](https://skll.readthedocs.io) machine learning library to train and evaluate the models.
+
+7. (Optional) **Evaluate trained model**. If you want to obtain detailed evaluation metrics for an RST parsing model, run:
+
+    ```bash
+    rst_eval rst_discourse_tb_edus_TRAINING_DEV.json -p rst_parsing_modelC1.0 --use_gold_syntax
+    ```
+
+    This command will compute precision, recall, and F1 scores for 3 scenarios: spans labeled with nuclearity and relation types, spans labeled only with nuclearity, and unlabeled token spans.  `--use_gold_syntax` means that the command will use gold standard EDUs and syntactic parses.
+
+    *NOTE*: While the evaluation script has basic functionality in place, at the moment it almost certainly does not appropriately handle important edge cases (e.g., same-unit relations, relations at the top of the tree). 
+
+### Use trained models
+
+At this point, we are ready to use the segmentation and RST parsing models to process raw text documents. To process a raw text document `document.txt` with the end-to-end parser (assuming `C` = 1.0 was the best hyper-parameter value for both the segmentation and RST parsing models), run:
+
+```bash
+rst_parse -g segmentation_model.C1.0 -p rst_parsing_model.C1.0 document.txt > output.json
 ```
-rst_eval rst_discourse_tb_edus_TRAINING_DEV.json -p rst_parsing_modelC1.0 --use_gold_syntax
+
+`output.json` contains a dictionary with two keys: `edu_tokens` and `scored_rst_trees`. The value corresponding to `edu_tokens` is a list of lists: each list is a list of tokens containing the segmented Elementary Discourse Units (EDUs). The value corresponding to `rst_trees` is a list of dictionaries: each dictionary has two keys, `tree` and `score` containing the RST parse tree for the document and its score respectively. By default, only a single tree is produed but additonal trees can be produced by specifying the `-n` option for `rst_parse`. 
```

Comment:
each list is a list of tokens containing the segmented Elementary Discourse Units (EDUs). -> each list is a list of tokens in a segmented Elementary Discourse Units (EDUs).

Reply:
Hmm, I think even this is not clear enough. Let me rewrite this a bit.